packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -47,7 +47,7 @@ export const vModelText: ModelDirective<
   HTMLInputElement | HTMLTextAreaElement
 > = {
   beforeMount(el, { value, modifiers: { lazy, trim, number } }, vnode) {
-    el.value = value
+    if (value !== undefined && value !== null) el.value = value
```

Comment:
you should also be able to do

```suggestion
    if (value != null) el.value = value
```